signature/src/error.rs

```diff
@@ -71,18 +71,17 @@ impl Display for Error {
     }
 }
 
-#[cfg(all(feature = "std", feature = "alloc"))]
-impl From<Box<dyn std::error::Error + Send + Sync + 'static>> for Error {
-    fn from(source: Box<dyn std::error::Error + Send + Sync + 'static>) -> Error {
+#[cfg(feature = "alloc")]
+impl From<Box<dyn core::error::Error + Send + Sync + 'static>> for Error {
+    fn from(source: Box<dyn core::error::Error + Send + Sync + 'static>) -> Error {
         Self::from_source(source)
     }
 }
 
-#[cfg(feature = "std")]
-impl std::error::Error for Error {
-    fn source(&self) -> Option<&(dyn std::error::Error + 'static)> {
+impl core::error::Error for Error {
+    fn source(&self) -> Option<&(dyn core::error::Error + 'static)> {
         self.source
             .as_ref()
-            .map(|source| source.as_ref() as &(dyn std::error::Error + 'static))
+            .map(|source| source.as_ref() as &(dyn core::error::Error + 'static))
```

Comment:
The issue is here -- since `source` only exists when the `alloc` feature is enabled, we need to gate this properly:
```suggestion
        #[cfg(feature = "alloc")]
        {
            self.source.as_ref()
            .map(|source| source.as_ref() as &(dyn core::error::Error + 'static))
        }
        #[cfg(not(feature = "alloc"))]
        {
            None
        }
```